File: survey/src/main/java/org/ksh/common/config/SwaggerConfig.java
package com.kata.boardkata.configration;

import io.swagger.v3.oas.models.OpenAPI;
import io.swagger.v3.oas.models.info.Info;
import io.swagger.v3.oas.models.servers.Server;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class SwaggerConfig {
	@Bean
	public OpenAPI openAPI() {

		Info info = new Info()
				.version("v1.0.0")
				.title("설문조사 API")
				.description("API Description");

		return new OpenAPI()
				.addServersItem(new Server().url("/"))
				.info(info);
	}

}

File: src/main/kotlin/org/example/survey/config/SwaggerConfig.kt
package org.example.survey.config

import io.swagger.v3.oas.models.Components
import io.swagger.v3.oas.models.OpenAPI
import io.swagger.v3.oas.models.info.Info
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.servlet.config.annotation.EnableWebMvc

@Configuration
@EnableWebMvc
class SwaggerConfig {
    @Bean
    fun openAPI(): OpenAPI =
        OpenAPI()
            .components(Components())
            .info(configurationInfo())

    private fun configurationInfo(): Info =
        Info()
            .title("Survey API")
            .description("Survey API Documentation")
            .version("1.0.0")
}
